Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -976,7 +976,9 @@ extension BrowserViewController: BrowserDelegate {
             let bars = self.snackBars.subviews
             // if the keyboard is showing then ensure that the snackbars are positioned above it, otherwise position them above the toolbar/view bottom
             if let state = keyboardState {
-                make.bottom.equalTo(-(state.height + 20))
+                if bars.count > 0 {
+                    make.bottom.equalTo(-(state.height + 20))
+                }
```

Comment:
Let's keep the `bottom` and `top` constraints together to help avoid similar mistakes in the future. I'd suggest grouping them like this:

```
if let state = keyboardState where bars.count > 0 {
    let view = bars[bars.count-1] as! UIView
    make.top.equalTo(view.snp_top)
    make.bottom.equalTo(-(state.height + 20))
} else {
    make.top.bottom.equalTo(self.toolbar?.snp_top ?? self.view.snp_bottom)
}
```